### PR TITLE
Vtk9

### PR DESCRIPTION
**Pull request by chraibi**
Code does not build properly with vtk9+qt5. 

These are some minimal changes to fix this build-issue

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -202,15 +202,15 @@ set( RCS
 
 
 #set(VTK_USE_HYBRID ON CACHE BOOL "doc" FORCE)
-find_package(VTK REQUIRED)
+find_package(VTK QUIET)
```
Comment (schroedtert) [inline CMakeLists.txt:205]: We should define a minimum version of VTK to compile.

Comment (schroedtert) [reply CMakeLists.txt:205]: Why `QUIET`, this will remove error messages if the package is not found?

File: CMakeLists.txt [hunk 2]
```diff
@@ -202,15 +202,15 @@ set( RCS
...
 
 if(VTK_FOUND)
   message(STATUS "Check VTK ..................................... yes (${VTK_VERSION})")
 else(VTK_FOUND)
   message(FATAL_ERROR "Check VTK ..................................... no")
+  return()
 endif(VTK_FOUND)
```
Comment (schroedtert) [inline CMakeLists.txt:212]: This will be done by `find_package( foo REQUIRED)`

File: CMakeLists.txt [hunk 3]
```diff
@@ -323,16 +321,17 @@ elseif( ${QT5_INSTALLED})
...
-    TARGET_LINK_LIBRARIES( jpsvis  ${VTK_LIBRARIES} ${QTLIBS}) #qt5
-  endif()
-else() # this else is not yet tested
-  target_link_libraries(jpsvis vtkHybrid QVTK vtkViews ${QTLIBS})
-endif()
+target_link_libraries(jpsvis ${VTK_LIBRARIES} Qt5::Widgets  Qt5::Network Qt5::Xml)
+if (VTK_VERSION VERSION_LESS "8.90.0")
+  # old system
+  include(${VTK_USE_FILE})
+else ()
```
Comment (schroedtert) [inline CMakeLists.txt:328]: We should discuss if we still support the old versions

File: CMakeLists.txt [hunk 4]
```diff
@@ -7,7 +7,7 @@
 #windows flags: -DDESIRED_QT_VERSION=5 -DCMAKE_PREFIX_PATH=D:\Qt32bit\Qt5.4.1\5.4\msvc2013 -DCMAKE_PREFIX_PATH=VTK.0.0\VTK-7.0.0\build -G "Visual Studio 12 2013"
 #--------------------------------------------------------------------------
 cmake_minimum_required(VERSION 3.2)
-project( jpsvis LANGUAGES CXX)
+project( jpsvis LANGUAGES CXX C)
```
Comment (schroedtert) [inline CMakeLists.txt:10]: This is needed for VTK right?

Comment (chraibi) [reply CMakeLists.txt:10]: Screen Shot 2021-09-02 at 09 53 11

File: CMakeLists.txt (hunk 1 above)
Comment (chraibi) [reply CMakeLists.txt:205]: > We should define a minimum version of VTK to compile.

Yes, but not in this PR

File: CMakeLists.txt (hunk 3 above)
Comment (chraibi) [reply CMakeLists.txt:327]: In our upcoming discussion I would suggest supporting vtk 8 upwards and qt 5 upwards.

**Comment by Ozaq** [comment]
David mentioned that you are working on this and I tried to see if I can get it to work. 
I am wondering if you can compile w.o. any additional changes?
I had to add `#include <vtkVersionMacros.h>` to all files doing a Vtk version check `#if VTK_MAJOR_VERSION <= 5`

* ThreadVisualisation.cpp
* TimerCallback.cpp
* TrailPlotter.cpp
* FacilityGeometry.cpp
* LinePlotter.cpp
* LinePlotter2D.cpp
* PointPlotter.cpp
* PointPlotter2D.cpp

**Comment by chraibi** [comment]
> David mentioned that you are working on this and I tried to see if I can get it to work.
> I am wondering if you can compile w.o. any additional changes?
> I had to add `#include <vtkVersionMacros.h>` to all files doing a Vtk version check `#if VTK_MAJOR_VERSION <= 5`
> 
>     * ThreadVisualisation.cpp
> 
>     * TimerCallback.cpp
> 
>     * TrailPlotter.cpp
> 
>     * FacilityGeometry.cpp
> 
>     * LinePlotter.cpp
> 
>     * LinePlotter2D.cpp
> 
>     * PointPlotter.cpp
> 
>     * PointPlotter2D.cpp
> 
> 
> EDIT: Ok nvm :) @chraibi just removed that in [3157502](https://github.com/JuPedSim/jpsvis/commit/3157502b3029378761fa1295a456a6f0c5ea9a4d)

@Ozaq I thought, we don't need to worry about VTK < 5 anyway! :-)
But, do you mean your solution helped fix build-problems?

**Comment by Ozaq** [comment]
@chraibi Including `vtkVersionMacros.h` fixed the compile errors since this is where `VTK_MAJOR_VERSION` is defined. But removing vtk5 support is much better :)

**Comment by chraibi** [comment]
> @chraibi Including `vtkVersionMacros.h` fixed the compile errors since this is where `VTK_MAJOR_VERSION` is defined. But removing vtk5 support is much better :)

Unrelated to this PR: 
it's a mystery to me why this compiled fine with vtk 8. 🤔

**Comment by chraibi** [comment]
fixes #81

**Comment by Ozaq** [comment]
> > @chraibi Including `vtkVersionMacros.h` fixed the compile errors since this is where `VTK_MAJOR_VERSION` is defined. But removing vtk5 support is much better :)
> 
> Unrelated to this PR:
> it's a mystery to me why this compiled fine with vtk 8. 🤔

I can only speculate but I would guess some changes in the VTK9 header files, i.e. JPSVis source included some vtk header that transitively included `vtkVersionMacros.h`.

**Review by schroedtert — APPROVED** [review]

**Pull request merged**